frontend/awx/access/credentials/CredentialForm.tsx

```diff
@@ -269,14 +268,124 @@ function CredentialInputs({
         }
       />
       {credentialTypeID && credentialTypes && credentialTypes[credentialTypeID] ? (
-        <CredentialSubForm credentialType={credentialTypes[credentialTypeID]} />
+        <CredentialSubForm
+          credentialType={credentialTypes[credentialTypeID]}
+          isEditMode={isEditMode}
+        />
       ) : null}
     </>
   );
 }
-function CredentialSubForm({ credentialType }: { credentialType: CredentialType | undefined }) {
-  const { t } = useTranslation();
 
+function PageFormSecretInput({
+  field,
+  isEditMode,
+  credentialType,
+  requiredFields,
+}: {
+  field: CredentialInputField;
+  isEditMode: boolean;
+  credentialType: CredentialType;
+  requiredFields: string[];
+}) {
+  const [t] = useTranslation();
+  const [shouldHideField, setShouldHideField] = useState(field.secret && isEditMode);
+  const { resetField } = useForm();
+  const [clear, setClear] = useState(false);
+  const { setValue } = useFormContext();
```

Comment:
I don't believe declaring resetField is necessary because if we remove ```const { resetField } = useForm();``` as suggested, then it won't be used in this function and causes an error.